src/charm.py

```diff
@@ -741,6 +713,11 @@ def check_pgb_running(self):
                 if self.unit.status.message != EXTENSIONS_BLOCKING_MESSAGE:
                     self.unit.status = BlockedStatus(pgb_not_running)
                 logger.warning(pgb_not_running)
+                if service == self._metrics_service:
+                    try:
+                        pgb_container.restart(service)
+                    except ChangeError as e:
+                        logger.debug(f"Failed to start metrics service with error: {e}")
```

Comment:
Try to restart the monitoring service if it is down. Going forward we should consider a more robust way to restart (notices?, observer?).